### ISSUE TITLE
[question] Action Smoothing

### ISSUE THREAD

**Issue by araffin** [rail-berkeley/softlearning#3]
Hello,
In the blog and in the code, action smoothing is mentioned but never explained...
In the code, one does apparently something like:
```python
# smoothing coeff (0 -> no smoothing, 1 -> max smoothing)
alpha = some_value
beta = sqrt(1 - alpha ** 2)
# raw latents is sampled from a MultivariateNormalDiag with zero mean and unit std
smoothing_latent = alpha * smoothing_latent + raw_latents
latent = beta * smoothing_latent

action = mean + std * latent
action = tanh(action)
```

My question is where does this smoothing comes from? and why not, for instance, an exponential smoothing?

Also, can `raw_latent` be seen as the noise?

**Comment by haarnoja** [comment]
Hi Antonin,

the smoothing is in fact currently wrong in the code. We are working to fix it. The correct form should be
```
beta = sqrt(1 - alpha ** 2) / (1 - alpha)**2
```
and 
```
smoothing_latent = alpha * smoothing_latent + (1-alpha) * raw_latents
```
which is exponential smoothing in the latent space. We prefer smoothing the latents instead of the actions because that does not introduce slowness, and the policy can react immediately to a change in  the state.  On the downside, if the observations (or state) is noisy, smoothing in the latent space can still result lots of noise in the action space.

**Comment by araffin** [comment]
Hi,
Thanks for your answer, that makes more sense.
Btw, i would like to congratulate you and your colleagues for this great algorithm. I have been using it recently on several projects (e.g. https://github.com/araffin/learning-to-drive-in-5-minutes) and it works pretty well with only minor hyperparameters tuning.

**Comment by hartikainen** [comment]
The action (latent) smoothing is fixed in #13.
